src/components/GoodList.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Options from './Options';
+
+class GoodList extends React.Component {
+  state = {
+    date: this.props.goodsFromServer,
```

Comment:
Мне кажется здесь уместнее назвать свойство **goodsList**
```suggestion
    goodsList: this.props.goodsFromServer,
```